src/main/java/bio/terra/service/duos/DuosClient.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.service.duos;
+
+import bio.terra.app.configuration.DuosConfiguration;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.duos.exception.DuosDatasetBadRequestException;
+import bio.terra.service.duos.exception.DuosDatasetNotFoundException;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class DuosClient {
+
+  private final DuosConfiguration duosConfiguration;
+  private final RestTemplate restTemplate;
+
+  @Autowired
+  public DuosClient(DuosConfiguration duosConfiguration, RestTemplate restTemplate) {
+    this.duosConfiguration = duosConfiguration;
+    this.restTemplate = restTemplate;
+    restTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+  }
+
+  /**
+   * @return a new unauthenticated HttpHeaders object for interacting with DUOS APIs
+   */
+  private HttpHeaders getHttpHeaders() {
+    HttpHeaders headers = new HttpHeaders();
+    headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+    return headers;
+  }
+
+  @VisibleForTesting
+  String getDatasetUrl(String duosId) {
+    return String.format("%s/api/tdr/%s", duosConfiguration.getBasePath(), duosId);
```

Comment:
Correct.  I've flagged our desire for one, but realistically even if they write one it will be awhile before it's made available:

https://broadinstitute.slack.com/archives/CKSCCUZPU/p1667922177959739